ui/src/app/subnets/views/VLANDetails/VLANSummary/VLANSummary.tsx

```diff
@@ -1,9 +1,122 @@
+import vlanSelectors from "app/store/vlan/selectors";
+import type { VLAN, VLANMeta } from "app/store/vlan/types";
+import subnetsURLs from "app/subnets/urls";
+import { breakLines } from "app/utils";
+
+type Props = {
+  id: VLAN[VLANMeta.PK] | null;
+};
+
+const VLANSummary = ({ id }: Props): JSX.Element | null => {
+  const vlan = useSelector((state: RootState) =>
+    vlanSelectors.getById(state, id)
+  );
+  const controllers = useSelector((state: RootState) =>
+    controllerSelectors.getByIDs(state, vlan?.rack_sids || [])
+  );
+  const controllersLoading = useSelector(controllerSelectors.loading);
+  const fabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, vlan?.fabric)
+  );
+  const fabricsLoading = useSelector(fabricSelectors.loading);
+  const space = useSelector((state: RootState) =>
+    spaceSelectors.getById(state, vlan?.space)
+  );
+  const spacesLoading = useSelector(spaceSelectors.loading);
+
+  if (!vlan) {
+    return null;
+  }
+
+  const fabricDisplay = getFabricDisplay(fabric);
+  const spaceDisplay = getSpaceDisplay(space);
+  return (
+    <Strip shallow>
+      <h2 className="p-heading--4">VLAN summary</h2>
+      <Row>
+        <Col size={6}>
+          <Definition label="VID" description={`${vlan.vid}`} />
+          <Definition label="Name" description={vlan.name} />
+          <Definition label="MTU" description={`${vlan.mtu}`} />
+          <Definition label="Description" description={vlan.description} />
+        </Col>
+        <Col size={6}>
+          <Definition label="Space">
+            {spacesLoading ? (
+              <Spinner />
+            ) : space ? (
+              <Link
+                data-testid="space-link"
+                to={subnetsURLs.space.index({ id: space.id })}
+              >
+                {spaceDisplay}
+              </Link>
+            ) : (
+              spaceDisplay
+            )}
+          </Definition>
+          <Definition label="Fabric">
+            {fabricsLoading ? (
+              <Spinner />
+            ) : fabric ? (
+              <Link
+                data-testid="fabric-link"
+                to={subnetsURLs.fabric.index({ id: fabric.id })}
+              >
+                {fabricDisplay}
+              </Link>
+            ) : (
+              fabricDisplay
+            )}
+          </Definition>
+          <Definition
+            label={
+              <>
+                Rack controllers
+                <Tooltip
+                  className="u-nudge-right--small"
+                  message={breakLines(
+                    "A rack controller controls hosts and images and runs network services like DHCP for connected VLANs."
+                  )}
+                >
+                  <Icon name="information" />
+                </Tooltip>
+              </>
+            }
+          >
+            {controllersLoading ? (
+              <Spinner />
+            ) : (
+              controllers.map((controller) =>
```

Comment:
It looks like this should be the primary and secondary racks: https://github.com/canonical-web-and-design/maas-ui/blob/master/legacy/src/app/partials/vlan-details.html#L379